lib/bibliotheca/guide.rb

```diff
@@ -44,6 +44,18 @@ def validate!
     def validate_slug!(a_slug)
       raise "inconsistent slug, must be #{slug}" if slug.present? && slug != a_slug
     end
+
+    def exercise_language(exercise)
+      exercise.language || self.language
+    end
+
+    def run_tests(exercise_id)
+      exercise = find_exercise_by_id(exercise_id)
+      raise Bibliotheca::Collection::ExerciseNotFoundError, "exercise #{exercise_id} not found" if exercise.nil?
```

Comment:
moves this logic to find_exercise_by_id broke 26 import tests. I think we should move it but in other branch